src/staggered_nonhydrostatic_model.jl

```diff
@@ -228,22 +237,17 @@ function dss!(Y, p, t)
 end
 
 function horizontal_limiter_tendency!(Yₜ, Y, p, t)
-    divₕ = Operators.Divergence()
-    C123 = Geometry.Covariant123Vector
-
-    (; ᶜu_bar) = p
-    (; ᶜinterp) = p.operators
-    ᶜuₕ = Y.c.uₕ
-    ᶠw = Y.f.w
-    @. ᶜu_bar = C123(ᶜuₕ) + C123(ᶜinterp(ᶠw))
-
     Yₜ .= zero(eltype(Yₜ))
+    set_precomputed_quantities!(Y, p, t)
+
+    (; ᶜu) = p
+    divₕ = Operators.Divergence()
```

Comment:
Why do we store `interp` in `p.operators`, but have to define `div_h` here? - Just asking to understand the logic on "what goes where"

Reply:
I agree, I think it would be cleaner to define them once for all ClimaAtmos.